core/src/main/java/com/datastrato/gravitino/catalog/PartitionNormalizeDispatcher.java

```diff
@@ -29,16 +26,23 @@ public PartitionNormalizeDispatcher(PartitionOperationDispatcher dispatcher) {
   public String[] listPartitionNames(NameIdentifier tableIdent) {
     String[] partitionNames =
         dispatcher.listPartitionNames(
-            CapabilityHelpers.applyCaseSensitive(tableIdent, Capability.Scope.TABLE, dispatcher));
-    return applyCaseSensitive(tableIdent, partitionNames);
+            applyCaseSensitive(tableIdent, Capability.Scope.TABLE, dispatcher));
+    return Arrays.stream(partitionNames)
+        .map(
+            partitionName ->
+                applyCaseSensitiveOnName(
+                    Capability.Scope.PARTITION,
+                    partitionName,
+                    dispatcher.getCatalogCapability(tableIdent)))
```

Comment:
We can extract `dispatcher.getCatalogCapability(tableIdent)` out of lambda to avoid multiple calls.

Reply:
fixed.